pkg/cmd/server/api/v1/types.go

```diff
@@ -138,6 +145,11 @@ type MasterConfig struct {
 
 	// NetworkConfig to be passed to the compiled in network plugin
 	NetworkConfig NetworkConfig `json:"networkConfig"`
+
+	// Enable or disable OpenShift components.
+	// TODO: find out how to write this field only if any of its field contains
```

Comment:
Make `FeatureConfig` a pointer and then `omitempty`.  Be sure to add comments on this field explaining that it is the exception to "never omitempty in our config", because disabling features should be uncommon and we don't want people to think that removing a "disable" will actually turn it on in the atomic case.